### PR TITLE
[crypto] expo-standard-web-crypto, a partial polyfill for the browser's crypto API

### PR DESCRIPTION
**Pull request by ide**
Why: expo-random now provides a way to synchronously generate random numbers, but packages that run on the web but not on Expo don't use that method. To provide easier compatibility with those packages, this commit introduces expo-web-crypto, a partial (very partial -- just one method is provided right now with no plans for more) polyfill for W3C Crypto.

How: The polyfill for Crypto.getRandomValues provides spec-compliant behavior. It validates the arguments and then calls into expo-random's getRandomBytes. The bytes are used to populate an underlying ArrayBuffer, which is then copied into the given destination TypedArray.

If expo-random's synchronous method is unavailable -- notably when using Chrome debugging -- an insecure implementation that uses Math.random is used. In general I expect this implementation rarely to be used as we'd just use the browser's `window.crypto` instead.

See README for usage.

Tests: Added comprehensive unit tests.

### CODE REVIEW

**Comment by ide** [comment]
CI looks good now.

File: packages/expo-standard-web-crypto/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+A partial implementation of the W3c Crypto API for Expo
+
+# API documentation
+
+This package provides a partial polyfill of the W3C Crypto API for Expo. Namely, `Crypto#getRandomValues()` is implemented. See the [W3C Crypto specification](https://www.w3.org/TR/WebCryptoAPI/) for the API documentation.
+
+# Installation in Expo projects
+
+First follow the instructions for installing `expo-random`, which is required by this package. Then install this package.
```
Comment (brentvatne) [inline packages/expo-standard-web-crypto/README.md:11]:
```suggestion
First follow the instructions for installing [`expo-random`](https://github.com/expo/expo/blob/master/packages/expo-random/README.md), which is required by this package. Then install this package.
```

File: packages/expo-standard-web-crypto/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+There are two ways to use this package: you can import a `Crypto` instance or you can globally define `crypto`. Some code may expect the latter.
+
+Importing a `Crypto` instance:
+```js
+import crypto from 'expo-standard-web-crypto';
+```
+
+Globally defining `crypto`:
+```js
```
Comment (brentvatne) [inline packages/expo-standard-web-crypto/README.md:29]:
```suggestion

```js
```

File: packages/expo-standard-web-crypto/README.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+```
+npm install expo-standard-crypto
+```
+
+### Using the polyfill
+
+There are two ways to use this package: you can import a `Crypto` instance or you can globally define `crypto`. Some code may expect the latter.
+
+Importing a `Crypto` instance:
+```js
```
Comment (brentvatne) [inline packages/expo-standard-web-crypto/README.md:24]:
```suggestion

```js
```

**Review by brentvatne — APPROVED** [review]

**Comment by brentvatne** [comment]
we should maybe document this in https://docs.expo.io/versions/latest/sdk/random/ also

**Comment by ide** [comment]
Added a sentence to the docs.

**Comment by brentvatne** [comment]
lgtm

**Pull request merged**